src/test/util/blocktools.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2019 The Unit-e developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/util/blocktools.h>
+
+namespace blocktools {
+
+CBlockIndex *BlockIndexFake::MakeBlockIndex(const uint256 &hash, CBlockIndex *const prev) {
+  auto result = block_indexes.emplace(hash, CBlockIndex());
+  CBlockIndex *const index = &result.first->second;
+  index->phashBlock = &result.first->first;
+  index->nHeight = prev ? prev->nHeight + 1 : 0;
+  index->pprev = prev;
+  return index;
+}
+
+CBlockIndex *BlockIndexFake::Generate(const std::size_t count, const uint256 &starting_point) {
+  auto starting_block = block_indexes.find(starting_point);
+  std::size_t height = 0;
+  CBlockIndex *const starting_index = [&]() {
+    if (starting_block == block_indexes.end()) {
+      ++height;
+      return MakeBlockIndex(starting_point, nullptr);
+    } else {
+      return &starting_block->second;
+    }
+  }();
+  assert(starting_index);
+  assert(starting_index->phashBlock);
+
+  CBlockIndex *current_index = starting_index;
+  for (; height < count; ++height) {
+    const uint256 hash = GetRandHash();
```

Comment:
I find it is very useful when writing and debugging tests, to use specific hashes instead of random ones. For example, you can choose hashes `[0, 1, 2, .. 50, .. 99]` for "main chain", `[0, 1, 2, ... 50, 10051, 10052, .. 10099]` for "fork1", and `[[0, 1, 2, ... 90, 20091, 20092, .. 20099]]` for "fork2". When things go wrong, random hashes in logs don't help you in debugging.

It would be nice to implement specific hashes here.

Reply:
Added in 56893e9ec9fd5f2ccadd1546d56b9b1dd5bf4289